src/app/tabs/newAdmin/addresses/operations.js

```diff
@@ -103,6 +122,57 @@ const setMultiChainAddress = (domain, chainId, address, isNew) => async (dispatc
   );
 };
 
+/**
+ * Sets values in the Definitive Resolver.
+ * @param {string} domain the domain to be set
+ * @param {uint8} chainId chain Id for the coin
+ * @param {address} address address to be set
+ * @param {bool} isNew is this a new address?
+ */
+const setDefinitiveAddress = (domain, chainId, address, isNew) => async (dispatch) => {
+  const chainName = getChainNameById(chainId);
+  const chainIndex = getIndexById(chainId);
+  const accounts = await window.ethereum.enable();
+  const currentAddress = accounts[0];
+
+  dispatch(requestSetChainAddress(chainName));
+
+  // encode value if it is not empty:
+  let encodeValue = address;
+  if (encodeValue !== EMPTY_ADDRESS) {
+    try {
+      encodeValue = formatsByCoinType[chainIndex].decoder(address);
+    } catch (error) {
+      return dispatch(errorSetChainAddress(chainName, ADDRESS_ENCODING_ERROR, address));
+    }
+  }
+
+  return definitiveResolver.methods.setAddr(namehash(domain), chainId, encodeValue)
+    .send({ from: currentAddress }, (error, result) => {
```

Comment:
Why are we using callback and not promise here?

Reply:
We handle transactions in both ways, as I'm working through the manager should I convert these all to promises?

Reply:
Let's leave it like this this time, and analyze how we should handle cb/promise in the ops one by one. Both approaches have pros and cons.